src/components/Select/SingleSelect.tsx

```diff
@@ -26,8 +26,8 @@ export const Select = ({
   );
   const [open, setOpen] = useState(false);
 
-  const onOpenChange = useCallback((newOpen?: boolean) => {
-    setOpen(open => newOpen ?? !open);
+  const onOpenChange = useCallback((open: boolean) => {
```

Comment:
this should be a toggle

Reply:
We use onOpenChange for the radix-based items like dropdown, context menu , tooltip and all
So in order to have consistency I added onOpenChange instead of onToggle